### ISSUE TITLE
[BUG] Model instability when moving one variable that was used as input in an LSTM model to target variable

### ISSUE THREAD

**Issue by prabhatbanjara** [neuralhydrology/neuralhydrology#256]
I am working with an LSTM model in two different versions. In the first version, I used reservoir storage as the input, and the model was quite stable. In the second version, I am using both reservoir storage and streamflow as target variables, but the model has become highly unstable. I am using the Nash–Sutcliffe Efficiency (NSE) as the loss function, and the training losses are fluctuating significantly. Any insights or suggestions to help address this issue would be greatly appreciated. Below I am attaching the loss function that is for the second case when using as 2 target variables.

<img width="832" height="499" alt="Image" src="https://github.com/user-attachments/assets/5f41326f-d238-4374-8fc5-92dc62f5e9c1" />

<img width="968" height="574" alt="Image" src="https://github.com/user-attachments/assets/658fda0d-2981-4790-9df7-492c9e41e89d" />

<img width="952" height="531" alt="Image" src="https://github.com/user-attachments/assets/5863a5e3-da65-4cca-80ff-82dd945ddc8b" />

<img width="963" height="567" alt="Image" src="https://github.com/user-attachments/assets/d2eca2bb-7bde-445c-a2a0-153976e94f10" />

<img width="967" height="544" alt="Image" src="https://github.com/user-attachments/assets/1991e1db-5c35-4e77-9b38-f526c0d19f68" />

<img width="969" height="516" alt="Image" src="https://github.com/user-attachments/assets/d8c65e21-4035-47d8-bc8d-56fb44c8edf2" />
